src/ol/interaction/Snap.js

```diff
@@ -288,25 +304,55 @@ class Snap extends PointerInteraction {
     return features;
   }
 
+  /**
+   * Checks if two snap data sets are equal.
+   * Compares the Y-coordinate of the `vertex` and the feature ID.
+   *
+   * @param {SnappedInfo} data1 The first snap data set.
+   * @param {SnappedInfo} data2 The second snap data set.
+   * @return {boolean} `true` if the data sets are equal, otherwise `false`.
+   * @private
+   */
+  areSnapDataEqual_(data1, data2) {
+    return (
+      data1.vertex[1] === data2.vertex[1] &&
```

Comment:
I think you'll want to check equality for the segment instead. I'd expect an `unsnap` event only when snapping leaves a segment, not whenever I move the cursor.
```suggestion
      data1.segment === data2.segment &&
```